File: src/main/java/epicsquid/superiorshields/capability/shield/CapabilityRegistry.java
package epicsquid.superiorshields.capability.shield;

import epicsquid.superiorshields.SuperiorShields;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.entity.player.Player;
import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.CapabilityManager;
import net.minecraftforge.common.capabilities.CapabilityToken;
import net.minecraftforge.common.util.LazyOptional;

public class CapabilityRegistry {

	public static final ResourceLocation SHIELD_CAP_ID = new ResourceLocation(SuperiorShields.MODID, "superior_shield");

	public static final Capability<IShieldCapability> SHIELD_CAPABILITY = CapabilityManager.get(new CapabilityToken<>() {
	});

	public static LazyOptional<IShieldCapability> getShield(final Player player) {
		return player.getCapability(SHIELD_CAPABILITY);
	}
}

File: src/main/kotlin/dev/epicsquid/superiorshields/registry/CapabilityRegistry.kt
package dev.epicsquid.superiorshields.registry

import dev.epicsquid.superiorshields.SuperiorShields
import dev.epicsquid.superiorshields.capability.SuperiorShieldCap
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.entity.LivingEntity
import net.minecraftforge.common.capabilities.Capability
import net.minecraftforge.common.capabilities.CapabilityManager
import net.minecraftforge.common.capabilities.CapabilityToken
import thedarkcolour.kotlinforforge.forge.getCapabilityOrThrow

object CapabilityRegistry {
	val SUPERIOR_SHIELD_CAP_ID = ResourceLocation(SuperiorShields.MODID, "superior_shield")

	val SUPERIOR_SHIELD_CAP: Capability<SuperiorShieldCap> =
		CapabilityManager.get(object : CapabilityToken<SuperiorShieldCap>() {})

	val LivingEntity.shield: SuperiorShieldCap
		get() = getCapabilityOrThrow(SUPERIOR_SHIELD_CAP)
}